### ISSUE TITLE
qstringview.h in CsCore has compilation problems

### ISSUE THREAD

**Issue by rwols** [copperspice/copperspice#64]
OS: macOS 10.13.2
compiler: tip-of-trunk clang
git hash: 58fe6ada4baada06cd139c8806e670f0e99943f4

```
[573/3513] Building CXX object src/core/CMakeFiles/CsCore1.4.dir/kernel/qmetatype.cpp.o
FAILED: src/core/CMakeFiles/CsCore1.4.dir/kernel/qmetatype.cpp.o
/usr/local/bin/clang++    @src/core/CMakeFiles/CsCore1.4.dir/kernel/qmetatype.cpp.o.rsp -MD -MT src/core/CMakeFiles/CsCore1.4.dir/kernel/qmetatype.cpp.o -MF src/core/CMakeFiles/CsCore1.4.dir/kernel/qmetatype.cpp.o.d -o src/core/CMakeFiles/CsCore1.4.dir/kernel/qmetatype.cpp.o -c ../../src/core/kernel/qmetatype.cpp
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
include/QtCore/qstringview.h:218:40: error: calling 'toCaseFolded' with incomplete return type 'QString8'
      return iter->toCaseFolded() == c.toCaseFolded();
                                     ~~^~~~~~~~~~~~~~
include/QtCore/qchar32.h:415:16: note: 'toCaseFolded' declared here
      QString8 toCaseFolded() const;
               ^
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
In file included from include/QtCore/qstringview.h:26:
include/QtCore/cs_string_view.h:32:39: error: incomplete type 'QString8' named in nested name specifier
      using const_iterator = typename S::const_iterator;
                                      ^~~
include/QtCore/qstringview.h:44:42: note: in instantiation of template class 'CsString::CsBasicStringView<QString8>' requested here
class Q_CORE_EXPORT QStringView : public CsString::CsBasicStringView<S>
                                         ^
include/QtCore/qstringview.h:299:14: note: in instantiation of template class 'QStringView<QString8>' requested here
      return QStringView8();
             ^
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
In file included from include/QtCore/qstringview.h:26:
include/QtCore/cs_string_view.h:33:39: error: incomplete type 'QString8' named in nested name specifier
      using iterator       = typename S::const_iterator;
                                      ^~~
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
In file included from include/QtCore/qstringview.h:26:
include/QtCore/cs_string_view.h:35:47: error: incomplete type 'QString8' named in nested name specifier
      using const_reverse_iterator = typename S::const_reverse_iterator;
                                              ^~~
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
In file included from include/QtCore/qstringview.h:26:
include/QtCore/cs_string_view.h:36:47: error: incomplete type 'QString8' named in nested name specifier
      using reverse_iterator       = typename S::const_reverse_iterator;
                                              ^~~
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
include/QtCore/qstringview.h:51:40: error: incomplete type 'QString8' named in nested name specifier
      using const_iterator  = typename S::const_iterator;
                                       ^~~
include/QtCore/qstringview.h:299:14: note: in instantiation of template class 'QStringView<QString8>' requested here
      return QStringView8();
             ^
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
include/QtCore/qstringview.h:52:40: error: incomplete type 'QString8' named in nested name specifier
      using iterator        = typename S::const_iterator;
                                       ^~~
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
include/QtCore/qstringview.h:54:47: error: incomplete type 'QString8' named in nested name specifier
      using const_reverse_iterator = typename S::const_reverse_iterator;
                                              ^~~
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
include/QtCore/qstringview.h:55:47: error: incomplete type 'QString8' named in nested name specifier
      using reverse_iterator       = typename S::const_reverse_iterator;
                                              ^~~
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
include/QtCore/qstringview.h:344:40: error: calling 'toCaseFolded' with incomplete return type 'QString8'
      return at(0).toCaseFolded() == c.toCaseFolded();
                                     ~~^~~~~~~~~~~~~~
include/QtCore/qchar32.h:415:16: note: 'toCaseFolded' declared here
      QString8 toCaseFolded() const;
               ^
../../src/core/kernel/qmetatype.h:303:7: note: forward declaration of 'QString8'
class QString8;
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
include/QtCore/qstring8.h:275:12: error: no matching constructor for initialization of 'CsString::CsString' (aka 'CsBasicString<CsString::utf8>')
         : CsString::CsString( str )
           ^                   ~~~
include/QtCore/cs_string.h:60:16: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const std::__1::allocator<unsigned char>' for 1st argument
      explicit CsBasicString(const A &a)
               ^
include/QtCore/cs_string.h:66:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >' for 1st argument
      CsBasicString(const CsBasicString &str) = default;
      ^
include/QtCore/cs_string.h:70:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >' for 1st argument
      CsBasicString(CsBasicString && str) = default;
      ^
include/QtCore/cs_string.h:94:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const char32_t *' for 1st argument
      CsBasicString(const char32_t *str, const A &a = A());
      ^
include/QtCore/cs_string.h:77:7: note: candidate template ignored: requirement 'std::is_same<QStringView<QString8>, const char *>::value || std::is_same<QStringView<QString8>, char *>::value' was not satisfied [with T = QStringView<QString8>]
      CsBasicString(const T &str, const A &a = A());
      ^
include/QtCore/cs_string.h:81:7: note: candidate template ignored: could not match 'const char [N]' against 'QStringView8' (aka 'QStringView<QString8>')
      CsBasicString(const char (&str)[N], const A &a = A());
      ^
include/QtCore/cs_string.h:108:7: note: candidate template ignored: could not match 'CsBasicStringView' against 'QStringView'
      CsBasicString(CsBasicStringView<U> str, const A &a = A());
      ^
include/QtCore/cs_string.h:87:7: note: candidate constructor template not viable: requires at least 2 arguments, but 1 was provided
      CsBasicString(const T &str, size_type size, const A &a = A());
      ^
include/QtCore/cs_string.h:92:7: note: candidate constructor template not viable: requires at least 2 arguments, but 1 was provided
      CsBasicString(const char (&str)[N], size_type size, const A &a = A());
      ^
include/QtCore/cs_string.h:112:7: note: candidate constructor template not viable: requires at least 3 arguments, but 1 was provided
      CsBasicString(CsBasicStringView<U> str, size_type indexStart, size_type size, const A &a = A());
      ^
include/QtCore/cs_string.h:116:7: note: candidate constructor template not viable: requires at least 2 arguments, but 1 was provided
      CsBasicString(Iterator begin, Iterator end, const A &a = A());
      ^
include/QtCore/cs_string.h:55:7: note: candidate constructor not viable: requires 0 arguments, but 1 was provided
      CsBasicString()
      ^
include/QtCore/cs_string.h:67:7: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
      CsBasicString(const CsBasicString &str, const A &a);
      ^
include/QtCore/cs_string.h:71:7: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
      CsBasicString(CsBasicString && str, const A &a);
      ^
include/QtCore/cs_string.h:97:7: note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
      CsBasicString(const CsBasicString &str, size_type indexStart, const A &a = A());
      ^
include/QtCore/cs_string.h:101:7: note: candidate constructor not viable: requires at least 2 arguments, but 1 was provided
      CsBasicString(size_type count, CsChar c, const A &a = A());
      ^
include/QtCore/cs_string.h:98:7: note: candidate constructor not viable: requires at least 3 arguments, but 1 was provided
      CsBasicString(const CsBasicString &str, size_type indexStart, size_type size, const A &a = A());
      ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
include/QtCore/qstring8.h:317:30: error: no matching member function for call to 'append'
         CsString::CsString::append(str, indexStart, numOfChars);
         ~~~~~~~~~~~~~~~~~~~~^~~~~~
include/QtCore/cs_string.h:162:22: note: candidate function not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >' for 1st argument
      CsBasicString &append(const CsBasicString &str, size_type indexStart, size_type size = npos);
                     ^
include/QtCore/cs_string.h:200:22: note: candidate template ignored: could not match 'CsBasicStringView' against 'QStringView'
      CsBasicString &append(CsBasicStringView<U> str, size_type indexStart, size_type size);
                     ^
include/QtCore/cs_string.h:168:22: note: candidate function template not viable: requires 2 arguments, but 3 were provided
      CsBasicString &append(Iterator begin, Iterator end);
                     ^
include/QtCore/cs_string.h:174:22: note: candidate function template not viable: requires 2 arguments, but 3 were provided
      CsBasicString &append(const T &str, size_type size);
                     ^
include/QtCore/cs_string.h:178:22: note: candidate function template not viable: requires 2 arguments, but 3 were provided
      CsBasicString &append(const char (&str)[N], size_type size);
                     ^
include/QtCore/cs_string.h:184:22: note: candidate function template not viable: requires single argument 'str', but 3 arguments were provided
      CsBasicString &append(const T &str);
                     ^
include/QtCore/cs_string.h:189:22: note: candidate function template not viable: requires single argument 'str', but 3 arguments were provided
      CsBasicString &append(const char (&str)[N]);
                     ^
include/QtCore/cs_string.h:196:22: note: candidate function template not viable: requires single argument 'str', but 3 arguments were provided
      CsBasicString &append(CsBasicStringView<U> str);
                     ^
include/QtCore/cs_string.h:165:22: note: candidate function not viable: requires 2 arguments, but 3 were provided
      CsBasicString &append(size_type count, CsChar c);
                     ^
include/QtCore/cs_string.h:161:22: note: candidate function not viable: requires single argument 'str', but 3 arguments were provided
      CsBasicString &append(const CsBasicString &str);
                     ^
include/QtCore/cs_string.h:164:22: note: candidate function not viable: requires single argument 'c', but 3 arguments were provided
      CsBasicString &append(CsChar c);
                     ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
include/QtCore/qstring8.h:458:40: error: no matching member function for call to 'find_fast'
            return CsString::CsString::find_fast(str, iter);
                   ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
include/QtCore/cs_string.h:241:22: note: candidate function not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >' for 1st argument
      const_iterator find_fast(const CsBasicString &str, const_iterator iter_begin) const;
                     ^
include/QtCore/cs_string.h:271:22: note: candidate function not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'CsString::CsChar' for 1st argument
      const_iterator find_fast(CsChar c, const_iterator iter_begin) const;
                     ^
include/QtCore/cs_string.h:260:22: note: candidate template ignored: requirement 'std::is_same<QStringView<QString8>, const char *>::value || std::is_same<QStringView<QString8>, char *>::value' was not satisfied [with T = QStringView<QString8>]
      const_iterator find_fast(const T &str, const_iterator iter_begin) const;
                     ^
include/QtCore/cs_string.h:268:22: note: candidate template ignored: could not match 'const char [N]' against 'QStringView8' (aka 'QStringView<QString8>')
      const_iterator find_fast(const char (&str)[N], const_iterator iter_begin) const;
                     ^
include/QtCore/cs_string.h:246:22: note: candidate function template not viable: requires 3 arguments, but 2 were provided
      const_iterator find_fast(const T &str, const_iterator iter_begin, size_type size) const;
                     ^
include/QtCore/cs_string.h:250:22: note: candidate function template not viable: requires 3 arguments, but 2 were provided
      const_iterator find_fast(const char (&str)[N], const_iterator iter_begin, size_type size) const;
                     ^
include/QtCore/cs_string.h:255:22: note: candidate function template not viable: requires single argument 'str', but 2 arguments were provided
      const_iterator find_fast(const T &str) const;
                     ^
include/QtCore/cs_string.h:264:22: note: candidate function template not viable: requires single argument 'str', but 2 arguments were provided
      const_iterator find_fast(const char (&str)[N]) const;
                     ^
include/QtCore/cs_string.h:240:22: note: candidate function not viable: requires single argument 'str', but 2 arguments were provided
      const_iterator find_fast(const CsBasicString &str) const;
                     ^
include/QtCore/cs_string.h:270:22: note: candidate function not viable: requires single argument 'c', but 2 arguments were provided
      const_iterator find_fast(CsChar c) const;
                     ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
include/QtCore/qstring8.h:461:42: error: no viable conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const QString8'
            return cs_internal_find_fast(str, iter);
                                         ^~~
include/QtCore/qstring8.h:274:7: note: candidate constructor
      QString8(QStringView8 str)
      ^
include/QtCore/qstring8.h:222:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const QString8 &' for 1st argument
      QString8(const QString8 &other) = default;
      ^
include/QtCore/qstring8.h:223:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'QString8 &&' for 1st argument
      QString8(QString8 &&other) = default;
      ^
include/QtCore/qstring8.h:225:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'QChar32' for 1st argument
      QString8(QChar32 c);
      ^
include/QtCore/qstring8.h:228:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const QChar32 *' for 1st argument
      QString8(const QChar32 *data, size_type numOfChars = -1)  {
      ^
include/QtCore/qstring8.h:252:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const QByteArray &' for 1st argument
      QString8(const QByteArray &str)
      ^
include/QtCore/qstring8.h:263:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const CsString::CsString &' (aka 'const CsBasicString<CsString::utf8> &') for 1st argument
      QString8(const CsString::CsString &other)
      ^
include/QtCore/qstring8.h:269:7: note: candidate constructor not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'CsString::CsString &&' (aka 'CsBasicString<CsString::utf8> &&') for 1st argument
      QString8(CsString::CsString &&other)
      ^
include/QtCore/qstring8.h:247:7: note: candidate template ignored: could not match 'const char [N]' against 'QStringView8' (aka 'QStringView<QString8>')
      QString8(const char (&cStr)[N])
      ^
include/QtCore/qstring8.h:858:60: note: passing argument to parameter 'str' here
      const_iterator cs_internal_find_fast(const QString8 &str, const_iterator iter_begin) const;
                                                           ^
include/QtCore/qstring8.h:468:40: error: no matching member function for call to 'rfind_fast'
            return CsString::CsString::rfind_fast(str, iter);
                   ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
include/QtCore/cs_string.h:274:22: note: candidate function not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >' for 1st argument
      const_iterator rfind_fast(const CsBasicString &str, const_iterator iter_end) const;
                     ^
include/QtCore/cs_string.h:273:22: note: candidate function not viable: requires single argument 'str', but 2 arguments were provided
      const_iterator rfind_fast(const CsBasicString &str) const;
                     ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
include/QtCore/qstring8.h:503:30: error: no matching member function for call to 'insert'
         CsString::CsString::insert(indexStart, str);
         ~~~~~~~~~~~~~~~~~~~~^~~~~~
include/QtCore/cs_string.h:463:22: note: candidate function not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >' for 2nd argument
      CsBasicString &insert(size_type indexStart, const CsBasicString &str);
                     ^
include/QtCore/cs_string.h:468:16: note: candidate function not viable: no known conversion from 'QString8::size_type' (aka 'long') to 'CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >::const_iterator' (aka 'CsStringIterator<CsString::utf8, std::__1::allocator<unsigned char> >') for 1st argument
      iterator insert(const_iterator posStart, CsChar c);
               ^
include/QtCore/cs_string.h:470:16: note: candidate function not viable: no known conversion from 'QString8::size_type' (aka 'long') to 'CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >::const_iterator' (aka 'CsStringIterator<CsString::utf8, std::__1::allocator<unsigned char> >') for 1st argument
      iterator insert(const_iterator posStart, const CsBasicString &str);
               ^
include/QtCore/cs_string.h:448:22: note: candidate template ignored: requirement 'std::is_same<QStringView<QString8>, const char *>::value || std::is_same<QStringView<QString8>, char *>::value' was not satisfied [with T = QStringView<QString8>]
      CsBasicString &insert(size_type indexStart, const T &str);
                     ^
include/QtCore/cs_string.h:452:22: note: candidate template ignored: could not match 'const char [N]' against 'QStringView8' (aka 'QStringView<QString8>')
      CsBasicString &insert(size_type indexStart, const char (&str)[N]);
                     ^
include/QtCore/cs_string.h:489:16: note: candidate template ignored: could not match 'CsBasicStringView' against 'QStringView'
      iterator insert(size_type indexStart, CsBasicStringView<U> str);
               ^
include/QtCore/cs_string.h:457:22: note: candidate function template not viable: requires 3 arguments, but 2 were provided
      CsBasicString &insert(size_type indexStart, const T &str, size_type size);
                     ^
include/QtCore/cs_string.h:461:22: note: candidate function template not viable: requires 3 arguments, but 2 were provided
      CsBasicString &insert(size_type indexStart, const char (&str)[N], size_type size);
                     ^
include/QtCore/cs_string.h:475:16: note: candidate function template not viable: requires 3 arguments, but 2 were provided
      iterator insert(const_iterator posStart, const T &str, size_type size);
               ^
include/QtCore/cs_string.h:479:16: note: candidate function template not viable: requires 3 arguments, but 2 were provided
      iterator insert(const_iterator posStart, const char (&str)[N], size_type size);
               ^
include/QtCore/cs_string.h:482:16: note: candidate function template not viable: requires 3 arguments, but 2 were provided
      iterator insert(const_iterator posStart, Iterator begin, Iterator end);
               ^
include/QtCore/cs_string.h:493:16: note: candidate function template not viable: requires at least 3 arguments, but 2 were provided
      iterator insert(size_type indexStart, CsBasicStringView<U> str, size_type srcStart, size_type srcSize = npos);
               ^
include/QtCore/cs_string.h:443:22: note: candidate function not viable: requires 3 arguments, but 2 were provided
      CsBasicString &insert(size_type indexStart, size_type count, CsChar c);
                     ^
include/QtCore/cs_string.h:469:16: note: candidate function not viable: requires 3 arguments, but 2 were provided
      iterator insert(const_iterator posStart, size_type count, CsChar c);
               ^
include/QtCore/cs_string.h:465:22: note: candidate function not viable: requires at least 3 arguments, but 2 were provided
      CsBasicString &insert(size_type indexStart, const CsBasicString &str,
                     ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
include/QtCore/qstring8.h:508:30: error: no matching member function for call to 'insert'
         CsString::CsString::insert(indexStart, str, srcStart, numOfChars);
         ~~~~~~~~~~~~~~~~~~~~^~~~~~
include/QtCore/cs_string.h:465:22: note: candidate function not viable: no known conversion from 'QStringView8' (aka 'QStringView<QString8>') to 'const CsString::CsBasicString<CsString::utf8, std::__1::allocator<unsigned char> >' for 2nd argument
      CsBasicString &insert(size_type indexStart, const CsBasicString &str,
                     ^
include/QtCore/cs_string.h:493:16: note: candidate template ignored: could not match 'CsBasicStringView' against 'QStringView'
      iterator insert(size_type indexStart, CsBasicStringView<U> str, size_type srcStart, size_type srcSize = npos);
               ^
include/QtCore/cs_string.h:448:22: note: candidate function template not viable: requires 2 arguments, but 4 were provided
      CsBasicString &insert(size_type indexStart, const T &str);
                     ^
include/QtCore/cs_string.h:452:22: note: candidate function template not viable: requires 2 arguments, but 4 were provided
      CsBasicString &insert(size_type indexStart, const char (&str)[N]);
                     ^
include/QtCore/cs_string.h:457:22: note: candidate function template not viable: requires 3 arguments, but 4 were provided
      CsBasicString &insert(size_type indexStart, const T &str, size_type size);
                     ^
include/QtCore/cs_string.h:461:22: note: candidate function template not viable: requires 3 arguments, but 4 were provided
      CsBasicString &insert(size_type indexStart, const char (&str)[N], size_type size);
                     ^
include/QtCore/cs_string.h:475:16: note: candidate function template not viable: requires 3 arguments, but 4 were provided
      iterator insert(const_iterator posStart, const T &str, size_type size);
               ^
include/QtCore/cs_string.h:479:16: note: candidate function template not viable: requires 3 arguments, but 4 were provided
      iterator insert(const_iterator posStart, const char (&str)[N], size_type size);
               ^
include/QtCore/cs_string.h:482:16: note: candidate function template not viable: requires 3 arguments, but 4 were provided
      iterator insert(const_iterator posStart, Iterator begin, Iterator end);
               ^
include/QtCore/cs_string.h:489:16: note: candidate function template not viable: requires 2 arguments, but 4 were provided
      iterator insert(size_type indexStart, CsBasicStringView<U> str);
               ^
include/QtCore/cs_string.h:443:22: note: candidate function not viable: requires 3 arguments, but 4 were provided
      CsBasicString &insert(size_type indexStart, size_type count, CsChar c);
                     ^
include/QtCore/cs_string.h:469:16: note: candidate function not viable: requires 3 arguments, but 4 were provided
      iterator insert(const_iterator posStart, size_type count, CsChar c);
               ^
include/QtCore/cs_string.h:463:22: note: candidate function not viable: requires 2 arguments, but 4 were provided
      CsBasicString &insert(size_type indexStart, const CsBasicString &str);
                     ^
include/QtCore/cs_string.h:468:16: note: candidate function not viable: requires 2 arguments, but 4 were provided
      iterator insert(const_iterator posStart, CsChar c);
               ^
include/QtCore/cs_string.h:470:16: note: candidate function not viable: requires 2 arguments, but 4 were provided
      iterator insert(const_iterator posStart, const CsBasicString &str);
               ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
include/QtCore/qstringview.h:141:49: error: call to non-static member function without an object argument
         return CsString::CsBasicStringView<S>::cbegin();
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
include/QtCore/qstring8.h:312:41: note: in instantiation of member function 'QStringView<QString8>::cbegin' requested here
         CsString::CsString::append(str.cbegin(), str.cend());
                                        ^
In file included from ../../src/core/kernel/qmetatype.cpp:31:
In file included from include/QtCore/qstring8.h:35:
include/QtCore/qstringview.h:157:49: error: call to non-static member function without an object argument
         return CsString::CsBasicStringView<S>::cend();
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
include/QtCore/qstring8.h:312:55: note: in instantiation of member function 'QStringView<QString8>::cend' requested here
         CsString::CsString::append(str.cbegin(), str.cend());
                                                      ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
[582/3513] Building CXX object src/core/CMakeFiles/CsCore1.4.dir/kernel/qabstractitemmodel.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by agserm** [comment]
Thanks for the report! I'm in the process of moving our OS X CI to support 10.13 right now. I will be checking this out tomorrow.

**Comment by bgeller** [comment]
Status update. This issue turns out to be happening on every build which uses clang and not just an OS X problem.  We have escalated this to our highest priority.  

We were a bit distracted the past two days when our local sewer company decided to backhoe and they ripped up everything. Took out cell, internet, and land lines for about 10 miles.  We saw about 20 AT&T trucks on site and I believe they were splicing like mad.  A bit scary to lose everything (including 911) just after the battle with net neutrality.
